File: 145-Binary-Tree-Postorder-Traversal.py
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def postorderTraversal(self, root):
        if root is None:
            return []

        return self.postorderTraversal(root.left) + self.postorderTraversal(root.right) + [root.val] 

File: 145-Binary-Tree-Postorder-Traversal.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        if (root == nullptr)
            return {};

        vector<int> res;
        traversal(res, root);
        return res;
    }

private:
    void traversal(vector<int>& res, TreeNode* root) {
        if (root == nullptr)
            return;

        traversal(res, root->left);
        traversal(res, root->right);
        res.push_back(root->val);
    }
};